sky/backends/onprem_utils.py

```diff
@@ -73,15 +75,20 @@ def check_and_get_local_clusters(suppress_error: bool = False) -> List[str]:
             user_config = yaml_config['auth']
             cluster_name = yaml_config['cluster']['name']
         sky_local_path = SKY_USER_LOCAL_CONFIG_PATH
-        if (AUTH_PLACEHOLDER
-                in (user_config['ssh_user'], user_config['ssh_private_key']) and
-                not suppress_error):
-            with ux_utils.print_exception_no_traceback():
-                raise ValueError('Authentication into local cluster requires '
-                                 'specifying `ssh_user` and `ssh_private_key` '
-                                 'under the `auth` dictionary. Please fill '
-                                 'aforementioned fields in '
-                                 f'{sky_local_path.format(cluster_name)}.')
+
+        if not suppress_error:
+            backend_utils.validate_schema(yaml_config,
+                                          schemas.get_cluster_schema(),
+                                          'Invalid cluster YAML: ')
```

Comment:
Should we do this validation _before_ L75 where we extract values from keys? Those lines may fail with KeyError if the yaml is malformed

Reply:
Done, thanks! Moved it to right after `yaml_config` is retrieved.